### PR TITLE
BIO1 2003 question 2

### PR DESCRIPTION
**Pull request by DaryaSanina**

### CODE REVIEW

**Comment by macaquedev** [comment]
oh and while I remember, can you change difficulty to Hard maybe?

File: solutions/bio1/Waves/index.tsx [hunk 1]
```diff
@@ -0,0 +1,74 @@
+import { SCodeBlock, SText, STitle, SList, SCode } from "components";
+
+export const Waves = () => {
+    return (
+        <>
+        <STitle>Observation</STitle>
+        <SText>
+          The main observation for this task is that the board size is small (only $9 \times 9$) and the number of pebbles is only $\leq 5$.
```
Comment (macaquedev) [inline solutions/bio1/Waves/index.tsx:8]: change to "the number of pebbles is at most 5"

File: solutions/bio1/Waves/index.tsx [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+          This allows us to iterate through all the cells and all the pebbles to calculate the contribution of each pebble to the depth in each cell.
+        </SText>
+
+        <STitle>Solution</STitle>
+        <SText>
+          For a given pebble and a given cell, we can calculate the contribution of the first wave of the pebble to the depth of the cell at time $r$.
+          Let the coordinates of the cell be $x_c$ and $y_c$, the coordinates of the pebble be $x_p$ and $y_p$, the time the pebble was thrown be $t_p$,
```
Comment (macaquedev) [inline solutions/bio1/Waves/index.tsx:15]: make it \$(x_c, y_c)$

File: solutions/bio1/Waves/index.tsx [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+          and the $x$-coordinates of the left and right banks be $x_l$ and $x_r$ respectively. Let's also call the time elapsed since the pebble was thrown $\Delta t = r - t_p$.
+        </SText>
+        <SText>
+          We can note that the time in which the wave reaches the cell if there are no collisions is $abs(x_c - x_p) + abs(y_c - y_p)$.
```
Comment (macaquedev) [inline solutions/bio1/Waves/index.tsx:19]: the time *at* which, not "in"

File: solutions/bio1/Waves/index.tsx [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+            The conditions for this case are $x_r \lt x_p \leq x_c$ and $\Delta t = x_c - x_p$.
+          </SText>
+          <SText>
+            The pebble was thrown in between the banks ($x_l \lt min(x_c, x_p)$ and $max(x_c, x_p) \lt x_r$).
+            We can add $x_p - x_l$ to $\Delta t$ and check how the right component of the wave would behave if it started at $x_l$.
+            The wave can be reflected either an odd or an even (possibly $0$) number of times.$\\$
+            For an even number of collisions, the wave should be able cover the distance between the banks an even number of times (returning to $x_l$) and then reach $x_c$,
+            so $\Delta t$ should be $\geq x_c - x_l$ and $\Delta t - (x_c - x_l)$ should be divisible by $2(x_r - x_l - 1)$.$\\$
+            For an odd number of collisions, the wave should be able to cover the distance between the banks an odd number of times (finishing at $x_r$) and then reach $x_c$,
+            so $\Delta t$ shold be $\geq (x_r - x_c) + (x_r - x_l - 1)$ and $\Delta t - (x_r - x_c) - (x_r - x_l - 1)$ should be divisible by $2(x_r - x_l - 1)$.
```
Comment (macaquedev) [inline solutions/bio1/Waves/index.tsx:55]: "should" not "shold"

**Review by macaquedev — CHANGES_REQUESTED** [review]
Just some really tiny points that could be better, but you've done a really amazing job, well done!! :)

**Review by macaquedev — APPROVED** [review]
thanks Daria, really good work!

**Review by DaryaSanina — COMMENTED** [review]
Ok, done

**Pull request merged**